hub/util/exceptions.py

```diff
@@ -304,13 +305,18 @@ def __init__(self, htype: str, key: str, available_keys: Sequence[str]):
         )
 
 
-class TensorMetaMismatchError(MetaError):
-    def __init__(self, meta_key: str, expected: Any, actual: Any):
-        super().__init__(
-            "Meta value for {} expected {} but got {}.".format(
-                meta_key, str(expected), str(actual)
-            )
-        )
+class TensorDtypeMismatchError(MetaError):
+    def __init__(self, expected: str, actual: str, htype: str):
+        msg = f"Dtype was expected to be '{expected}' instead it was '{actual}'. If you called `create_tensor` explicitly with `dtype`, your samples should also be of that dtype."
+
+        # TODO: we may want to raise this error at the API level to determine if the user explicitly overwrote the `dtype` or not. (to make this error message more precise)
```

Comment:
this can be another task, the current message i have is sufficient, just added this TODO because it can be better

Reply:
if anyone disagrees, i can do this now. let me know

Reply:
Maybe it will be better to have a separate check for scalars. Tried with class_labels and got `Dtype was expected to be 'uint32' instead it was 'int64'`
```
ds = hub.Dataset("./local_ds")
ds.create_tensor("label", "class_label")
ds.label.append(3)
```

Reply:
+1. Just ran into this.

Reply:
will open a separate PR for this